OutOfSchool/OutOfSchool.DataAccess/Repository/PhotoRepository.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+
+namespace OutOfSchool.Services.Repository
+{
+    public class PhotoRepository : IPhotoRepository
+    {
+        private readonly string basePhotoPath;
+
+        public PhotoRepository(IConfiguration config)
+        {
+            basePhotoPath = config.GetSection("PhotoSettings:BasePath").Value;
+        }
+
+        /// <summary>
+        /// Create photo.
+        /// </summary>
+        /// <param name="photo">Photo as byte array.</param>
+        /// <param name="fileName">Photo name.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task CreateUpdatePhoto(byte[] photo, string fileName)
+        {
+            var filePath = GenerateFilePath(fileName);
+
+            Directory.CreateDirectory(Path.GetDirectoryName(filePath));
+
+            await File.WriteAllBytesAsync(filePath, photo).ConfigureAwait(false);
+        }
```

Comment:
public Task CreateUpdatePhoto(byte[] photo, string fileName)
        {
            var filePath = GenerateFilePath(fileName);

            Directory.CreateDirectory(Path.GetDirectoryName(filePath));

            return File.WriteAllBytesAsync(filePath, photo);
        }
you can avoid unnecessary await here and just return task that will be awaited somewhere higher))